server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/MockMessageHandler.java

```diff
@@ -57,29 +55,25 @@
         @Override
         public void behave(T input) throws RejectException {
             Response response = behavior.getResponse();
-            if (response.isServerRejected()) {
-                throw new RejectException(response.getCode().getRawCode(), response.getMessage());
-            }
-            throw new NotImplementedException("Not rejecting commands in mock behaviours is not supported yet");
+            throw new RejectException(response.getCode().getRawCode(), response.getMessage());
         }
     }
 
-    class SMTPBehaviorRepositoryUpdater<T> implements Behavior<T> {
+    static class BehaviorDecorator<T> implements Behavior<T> {
+        private final Behavior<T> behavior;
+        private final Runnable decorator;
```

Comment:
Add something more precise than a `Runnable`

Reply:
It maybe should more be called a `callback` (variable name)

I tend to really like the generic aspect of `Runnable` type, and avoid nesting business responsabilities that relies in top level objects / methods into **inner classes** not part of the API.

Proposal: make these class private, hence the technical typing chosen is not exposed in any API.

Would you agree?